code/model/VirtualPage.php

```diff
@@ -413,7 +413,7 @@ public function CMSTreeClasses($numChildrenMethod="numChildren") {
 	public function __get($field) {
 		if(parent::hasMethod($funcName = "get$field")) {
 			return $this->$funcName();
-		} else if(parent::hasField($field)) {
+		} else if(parent::hasField($field) || ($field === 'ID' && !$this->getField('ID'))) {
```

Comment:
Oh, awesome. I’d avoided using `inInDB()` because that triggers the magic lookup I’m trying to avoid, but `exists()` uses `$this->record['ID']` directly so shouldn’t suffer from the same issue